src/Regulator.php

```diff
@@ -325,6 +327,138 @@ protected function dispatch_callback( Task $task, int $delay ): void {
 		}
 	}
 
+	/**
+	 * Run a set of tasks.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @param Task[] $tasks     The tasks to run.
+	 * @param array  $callables The callables to run.
+	 *
+	 * @return void
+	 */
+	public function run( array $tasks, array $callables = [] ): void {
+		$prefix = Config::get_hook_prefix();
+
+		if ( ! did_action( "shepherd_{$prefix}_tables_registered" ) ) {
+			foreach ( $tasks as $task ) {
+				$task->process();
+
+				/**
+				 * Fires an action when a task is run synchronously.
+				 *
+				 * @since 0.1.0
+				 *
+				 * @param Task $task The task that was dispatched synchronously.
+				 */
+				do_action( "shepherd_{$prefix}_task_run_sync", $task );
+			}
+
+			return;
+		}
+
+		if ( did_action( 'action_scheduler_init' ) || doing_action( 'action_scheduler_init' ) ) {
+			$this->run_callback( $tasks, $callables );
+			return;
+		}
+
+		add_action(
+			'action_scheduler_init',
+			function () use ( $tasks, $callables ): void {
+				$this->run_callback( $tasks, $callables );
+			},
+			10
+		);
+	}
+
+	/**
+	 * Runs a set of tasks.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @param Task[] $tasks     The tasks to run.
+	 * @param array  $callables The callables to run.
+	 *
+	 * @return void
+	 */
+	private function run_callback( array $tasks, array $callables = [] ): void {
+		/** @var array{before: callable( Task $task ): void, after: callable( Task $task ): void, always: callable( list<Task> $tasks ): void} $callables */
+		$callables = wp_parse_args(
+			$callables,
+			[
+				'before' => static function ( Task $task ): void {},
+				'after'  => static function ( Task $task ): void {},
+				'always' => static function ( array $tasks ): void {},
+			]
+		);
+
+		$context = defined( 'WP_CLI' ) && WP_CLI ? ' CLI' : '';
+		$context = ! $context && defined( 'REST_REQUEST' ) && REST_REQUEST ? ' REST' : $context;
+		$prefix  = Config::get_hook_prefix();
+
+		$runner = ActionScheduler_QueueRunner::instance();
+
+		/**
+		 * Filters the number of tasks to clean up after.
+		 *
+		 * @since 0.1.0
+		 *
+		 * @param int $clean_up_memory_every The number of tasks to clean up the memory after.
+		 */
+		$clean_up_memory_every = apply_filters( "shepherd_{$prefix}_clean_up_memory_every", 10 );
+
+		foreach ( array_values( $tasks ) as $offset => $task ) {
+			if ( ! in_array( $task->get_id(), $this->scheduled_tasks, true ) ) {
+				$this->dispatch_callback( $task, 0 );
+			}
+
+			if ( is_callable( $callables['before'] ) ) {
+				$callables['before']( $task );
```

Comment:
Here, and for all other callables, add error handling on exception thrown by the callable.
Add tests to cover it.
If not already handled (I think it is), do the same to handle an error thrown during the execution of the main task in the line:
```php
$runner->process_action( $task->get_action_id(), "Shepherd{$context}" );
```